### PR TITLE
Share image phase1

### PR DESCRIPTION
**Pull request by ozhe**
No AJAX and Bootstrap modal added.

### CODE REVIEW

File: test/mailers/previews/image_mailer_preview.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Preview all emails at http://localhost:3000/rails/mailers/image_mailer
+class ImageMailerPreview < ActionMailer::Preview
+end
```
Comment (HaominZhang) [inline test/mailers/previews/image_mailer_preview.rb]: I think we can have a method here to call the method you created in `image_mailer.rb`

File: app/mailers/image_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class ImageMailer < ApplicationMailer
+  default from: 'do_not_reply@ozhe.com'
+
+  def share_image_email(email_address:, image_id:, message: nil)
+    @email_address = email_address
+    @image = Image.find(image_id)
```
Comment (HaominZhang) [inline app/mailers/image_mailer.rb]: I think in your controller method we already find the image, do we need to pass in the id and find it again?

File: Gemfile [hunk 3]
```diff
@@ -30,6 +30,8 @@ gem 'simple_form'
 gem 'acts-as-taggable-on', '~> 4.0'
 # Add Bootstrap 4
 gem 'bootstrap', '~> 4.0.0.alpha5'
+# Use validates_email_format_of
+gem 'validates_email_format_of'
```
Comment (HaominZhang) [inline Gemfile]: Did we previously have an issue to validate the format of email address? Can we reuse that logic?

File: app/forms/share_image_form.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class ShareImageForm
```
Comment (HaominZhang) [inline app/forms/share_image_form.rb]: Do we need to add some unit tests for this form object?

File: test/flows/images_share_test.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
+require 'flow_test_helper'
+
+class ImagesCrudTest < FlowTestCase
```
Comment (HaominZhang) [inline test/flows/images_share_test.rb]: Should this class name be `ImagesShareTest`?

File: Gemfile (hunk 3 above)
Comment (ozhe) [inline Gemfile]: @HaominZhang We have the validator for url but not for email address.

Comment (HaominZhang) [inline Gemfile]: Oh my bad. Bad memory :(

**Pull request merged**